pkg/controller/clusters/status.go

```diff
@@ -85,6 +86,13 @@ func UpdateClusterStatusForCNI(ctx context.Context, cluster *anywherev1.Cluster)
 // updateConditionsForEtcdAndControlPlane updates the ControlPlaneReady condition if etcdadm cluster is not ready.
 func updateConditionsForEtcdAndControlPlane(cluster *anywherev1.Cluster, kcp *controlplanev1.KubeadmControlPlane, etcdadmCluster *etcdv1.EtcdadmCluster) {
 	// Make sure etcd cluster is ready before marking ControlPlaneReady status to true
+	// This condition happens while creating a workload cluster from the management cluster using controller
+	// where it tries to get the etcdadm cluster for the first time before it generates the resources.
+	if cluster.Spec.ExternalEtcdConfiguration != nil && etcdadmCluster == nil {
+		conditions.MarkFalse(cluster, anywherev1.ControlPlaneReadyCondition, anywherev1.RollingUpgradeInProgress, clusterv1.ConditionSeverityInfo, "Etcd cluster is not available yet")
```

Comment:
This could this not happen on create as well right? Is RollingUpgradeInProgress the right reason here?


Maybe we should create a new reason for these etcd scenarios. E.g. `anywherev1.ExternalEtcdNotReady`